lib/services/android-project-service.ts

```diff
@@ -196,13 +201,58 @@ class AndroidProjectService implements IPlatformProjectService {
 		}
 	}
 
-	private validateAndroidTarget(frameworkDir: string) {
+	private validateAndroidTarget(frameworkDir: string): string {
 		var validTarget = this.getTarget(frameworkDir).wait();
-		var output = this.$childProcess.exec('android list targets').wait();
-		if (!output.match(validTarget)) {
-			this.$errors.fail("Please install Android target %s the Android newest SDK). Make sure you have the latest Android tools installed as well. Run \"android\" from your command-line to install/update any missing SDKs or tools.",
-				validTarget.split('-')[1]);
+		var installedTargets = this.getInstalledTargets().wait();
+		var newTarget: string = undefined;
+		var match = _.contains(installedTargets, validTarget);
+		if (!match) {
+			// adjust to the latest available version
+			newTarget = this.getCompatibleTarget();
+			if (!newTarget) {
+				this.$errors.fail("Please install Android target %s. Make sure you have the latest Android tools installed as well." +
+									" Run \"android\" from your command-line to install/update any missing SDKs or tools.",
+									validTarget.split('-')[1]);
+			}
 		}
+
+		return newTarget;
+	}
+
+	private getCompatibleTarget(): string {
+		var installedTargets = this.getInstalledTargets().wait();
+		var compatibleTarget = _(this.SUPPORTED_TARGETS).sort().reverse().find(supportedTarget => _.contains(installedTargets, supportedTarget));
```

Comment:
Maybe you can use findLast instead of reverse and find
 https://lodash.com/docs#findLast